.pre-commit-config.yaml

```diff
@@ -42,4 +42,4 @@ repos:
       - id: no-commit-to-branch
         name: Prevent Commit to Main Branch
         args: ["--branch", "main"]
-        stages: [pre-commit]
+        stages: [commit]
```

Comment:
I am pretty sure this should be `pre-commit`.  Why do you think it should be `commit`?

Reply:
From `pre-commit.com`: "new in 3.2.0: The values of stages match the hook names. Previously, commit, push, and merge-commit matched pre-commit, pre-push, and pre-merge-commit respectively."

We should be using `pre-commit` as the name of the hook as we have the `pre-commit.ci` auto-updating the `pre-commit` version.

Reply:
I had to edit it since for some reason `pre-commit run` was failing due to the name being `pre-commit` instead of `commit`. Not sure if this problem is present for anyone else?

Reply:
this is a workaround not a fix presumably, so let's try and figure out why this is happening and do a proper fix.  Always aim to make code better, not make code run.

Reply:
I reverted it. Though it might be due to local configs, I still get this error:
```
An error has occurred: InvalidConfigError: 
==> File .pre-commit-config.yaml
==> At Config()
==> At key: repos
==> At Repository(repo='https://github.com/pre-commit/pre-commit-hooks')
==> At key: hooks
==> At Hook(id='no-commit-to-branch')
==> At key: stages
==> At index 0
=====> Expected one of commit, commit-msg, manual, merge-commit, post-checkout, post-commit, post-merge, post-rewrite, prepare-commit-msg, push but got: 'pre-commit'
```

Reply:
make sure your local pre-commit is up to date?